src/multi.c

```diff
@@ -63,12 +63,14 @@ bool MultiHandleCommand(RedisRaftCtx *rr,
         return true;
     } else if (cmd_len == 4 && !strncasecmp(cmd_str, "EXEC", 4)) {
         if (!multiState->active) {
+            setAskingState(rr, ctx, false);
             RedisModule_ReplyWithError(ctx, "ERR EXEC without MULTI");
             return true;
```

Comment:
I have the feeling we abuse the return value - in some cases, like in `EXEC` - false means - "cmds is ready and we can call`handleRedisCommandAppend()`.  in other cases - false means = "no MULTI-EXEC op, continue to handle the command in `handleRedisCommandAppend()`. True sometimes means Error, or sometimes expected behavior. Resetting the asking state is also not so clear - sometimes on errors we reset it, and sometimes not (like in the case of unsupported commands in the middle of MULTI-EXEC). 
Understand the logic, but it make it hard to follow this function. Suggest rewriting it to be more readable. 🙏

Reply:
I don't see that.  there are only 2 return falses

1) on exec
2) when it has nothing to do with multi (i.e. we didn't do anything at all).  every other case is return true.